gen3/Intro.py

```diff
@@ -0,0 +1,205 @@
+from gen3.submission import Gen3Submission 
+from gen3.analysis import Gen3Analysis 
+auth = Gen3Auth(api, refresh_file=cred)
+sub = Gen3Submission(api, auth) 
+
+​analysis = Gen3Analysis(api, auth) 
+
+!wget https://raw.githubusercontent.com/cgmeyer/gen3sdk-python/master/expansion/expansion.py
+%run expansion.py 
+exp = Gen3Expansion(api, auth) 
+query_txt = """ { subject (first:0) {project_id submitter_id id index_date} }"""
+query_data = sub.query(query_txt) 
+projects = exp.get_project_ids()
+
+####################################################################
+
+#use this
+
+####################################################################
+api = 'https://gen3.datacommons.io/' 
+cred = '/Users/bowenbao/Documents/Project/credentials.json'
+import pandas as pd
+import sys
+project_dir = "/Users/bowenbao/Documents/Project/"
+sys.path.insert(1, '{}'.format(project_dir))
+from gen3.auth import Gen3Auth 
+from gen3.submission import Gen3Submission 
+#from gen3.analysis import Gen3Analysis 
+auth = Gen3Auth(api, refresh_file=cred)
+sub = Gen3Submission(api, auth) 
+%run /Users/bowenbao/Documents/Project/gen3/analysis.py
+
+​analysis = Gen3Analysis(api, auth) 
+
+!wget https://raw.githubusercontent.com/cgmeyer/gen3sdk-python/master/expansion/expansion.py
+%run expansion.py 
+exp = Gen3Expansion(api, auth) 
+query_txt = """ { subject (first:0) {project_id submitter_id id index_date} }"""
+query_data = sub.query(query_txt) 
+projects = exp.get_project_ids()
+
+#not in ipython 
+#change manually, ipython default still in bowenbao
+os.chdir("Documents/Project")
+#df = pd.read_csv('lab_test.tsv',sep='\t') 
+df = pd.read_csv('lab_test2.tsv',sep='\t') 
+
+#re-run 
+%run /Users/bowenbao/Documents/Project/gen3/analysis.py
+​analysis = Gen3Analysis(api, auth) 
+
+dir(analysis) 
+
+df.dtypes
+
+df.dtypes["activity_area"]
+
+###Testing 
+
+analysis.plot_categorical_property(property = "sample_composition",df = df) 	#works
+
+analysis.plot_categorical_property_by_order(property = "sample_composition",df = df) #works
+
+analysis.plot_numeric_property(df =df,property = "activity_area") #works
+
+#needs work - how to group by project
+analysis.plot_numeric_property(df =df,property = "activity_area", by_project=True) #only one project #needs work
+
+#needs work
+analysis.node_record_counts(project_id = "OpenAccess-CCLE")	#query_txt = """{node (first:-1, project_id:"%s"){type}}""" problem 
+
+analysis.property_counts_table("sample_composition", df) #works 
+
+analysis.property_counts_by_project("sample_composition", df) #works
+
+#table is not defined - has issue importing 
+analysis.save_table_image(df,filename='mytable.png') 
+
+#new
+analysis.plotviolin_numeric_property_by_categorical_property(df =df, numeric_property = "activity_area", categorical_property = "sample_composition")
+
+analysis.pie_categorical_property_count(property = "sample_composition", df = df)
+
+analysis.scatter_numeric_by_numeric(df, "activity_area", "max_activity")
```

Comment:
The scatter plots are nice. A cool feature to add to this would be an optional categorical variable used to color-code the dots based on the values.  A simple example would be this "max_activity vs. activity_area" plot could have gender color coding, blue = male and red = female.  Probably best if you use some automatic coloring scheme (I prefer color-blind friendly palettes) that would pick colors for you given any number of categories.  The plot will also need a legend for the coloring.